src/algorithm/nest_cfgs.rs

```diff
@@ -428,8 +428,14 @@ pub(crate) mod test {
         //               \-> right -/             \-<--<-/
         let mut cfg_builder = CFGBuilder::new(type_row![NAT], type_row![NAT])?;
 
-        let pred_const = cfg_builder.add_constant(ConstValue::simple_predicate(0, 2))?; // Nothing here cares which
-        let const_unit = cfg_builder.add_constant(ConstValue::simple_unary_predicate())?;
+        let pred_const = cfg_builder.add_constant(
```

Comment:
Is it worth making `add_constant` take a tuple, so that we can call one factory method (thinking of the case of predicates here) that returns both value and type together?

Reply:
this suggests there is utility in a `TypedValue` that holds both.... Could make it a builder-only concept

Reply:
`add_load_const` too